### PR TITLE
Add new input field called domain, with a feature flag

### PR DESCRIPTION
**Pull request by emekafredy**
Add a new input field name `Domain` which has options on which team a slack bug ticket fits into

#### Jira Ticket
- [Link](https://syftapp.atlassian.net/browse/PTE-11801)

### CODE REVIEW

**Comment by keram** [comment]
Hi @emekafredy cc @faisal-raza1208 ,
Overall the PR looks good. 
Unfortunately we made small mistake on Friday which will require some changes but it should not be complicated.
The new changes should be related to only IndeedFlex (Syft) support bot instance and
not to affect the default configuration.
That means we will need to:

 - discard changes in `support.showForm`
 - remove src/views/support/default/bug_domain_feature.json
 - update configs.syft.view method to return the feature flagged view. Something like we had in the `showForm`
    https://github.com/Syft-Application/podpora-bot/blob/master/src/lib/support_config.ts#L122-L124
   ```js
       view: function(key: string): View {
        let template_name = key;
        if (feature.is_enabled('bug_report_with_domain_select_box_feature') && key === 'bug') {
            template_name = 'bug_domain_feature'
        }

        return views.support.syft[template_name];
    },
   ```
  - update `configs.syft.messageText` to include the `domain` in the description if and only if the feature
    flag is enabled       

For visual changes it is good also provide some screenshots. From the slack modal preview:
Screenshot 2023-03-06 at 11 23 46

Which brings me to last point. I see for the title we used name `Product Area`. It will be good if keys in code match the user names so we should rename `domain` to `product_area`.
Similarly for the feature flag bit more expressive name. For example: `bug_report_with_product_area_select_box`

**Comment by emekafredy** [comment]
This has been updated @keram

**Comment by kieran-flex** [comment]
The "Product Area" label and "domain" key was kind of deliberate in that I'd asked the guys to do that, but in hindsight, but it's not the right way to do it. I was trying to make a distinction between us asking the end user about "Product Area" as that is something people will understand, but under the hood, mapping it to "Domain" as that is how PTE teams are structured and how we'll do assignment.

File: src/lib/support.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ const support = {
 
         const config = supportConfig(support.configName(slack));
         const view: View = config.view(request_type);
+
```
Comment (keram) [inline src/lib/support.ts:47]: can this new line be removed please

File: src/lib/support_config.ts [hunk 2]
```diff
@@ -186,11 +191,17 @@ Submitted by: ${user.name}`;
         request_type: RequestType
     ): string {
         if (request_type === 'bug') {
+            let product_area_submission = ''
```
Comment (keram) [inline src/lib/support_config.ts:194]: Hi @emekafredy 
Have you installed pre-commit hooks using the `overcommit` tool?
If yes, the commit should fail with error like this:

Screenshot 2023-03-07 at 10 35 33

I really recommend using the pre-commit hooks as it save time.
I will add the linter check also to CI.
Meanwhile can you please run on this branch `npx eslint "**/*.ts" --fix` and address the reported warnings and errors?

File: src/lib/support_config.ts [hunk 3]
```diff
@@ -120,7 +120,12 @@ configs.syft = {
         return commandsHelpText(this.commands);
     },
     view: function(key: string): View {
-        return views.support.syft[key];
+        let template_name = key;
+        if (feature.is_enabled('bug_report_with_product_area_select_box') && key === 'bug') {
+            template_name = 'bug_domain_feature'
```
Comment (keram) [inline src/lib/support_config.ts:125]:
```suggestion
            template_name = 'bug_report_with_product_area'
```

File: src/views/support/syft/bug_domain_feature.json [hunk 4]
```diff
@@ -0,0 +1,420 @@
+{
```
Comment (keram) [inline src/views/support/syft/bug_domain_feature.json:1]: Would suggest to rename 
`bug_domain_feature.json` to `bug_report_with_product_area.json` to be consistent.

**Review by keram — CHANGES_REQUESTED** [review]

**Comment by emekafredy** [comment]
updated @keram

**Review by keram — APPROVED** [review]

**Comment by keram** [comment]
Thank you @emekafredy as next step please prepare draft PR that will remove the feature flag and make the new feature being normal part of the code.

**Pull request merged**